src/Symfony/Component/Notifier/Recipient/Recipient.php

```diff
@@ -13,30 +13,34 @@
 
 /**
  * @author Fabien Potencier <fabien@symfony.com>
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
  *
  * @experimental in 5.0
  */
-class Recipient
+class Recipient implements EmailRecipientInterface, SmsRecipientInterface
 {
+    use EmailValidationTrait;
+    use PhoneValidationTrait;
+
     private $email;
+    private $phone;
 
-    public function __construct(string $email = '')
+    public function __construct(string $email, string $phone)
```

Comment:
Making both email and phone required is not possible to enforce. The current AdminRecipient does not enforce it either. It's perfectly valid to have an admin/recipient without a phone or with only a phone number and no email.

Reply:
Oh yes that's true. How about introducing a check to ensure that at least one of email or phone is not empty? Basically something like this:

```
    public function __construct(string $email = '', string $phone = '')
    {
        if (empty(trim($email)) && empty(trim($phone))) {
            throw new LogicException(sprintf('"%s" needs an email or a phone but both cannot be empty.', static::class));
        }
```

Reply:
I am not sure I agree here. Making both attributes optional means that I couldn't rely on the interface being implementing when passing recipients around and thus defeating their purpose.

Reply:
@xabbuh 

I think having `has*()` methods in the Interfaces would solve this issue. We could either make `$email` and `$phone` required or optional and userland code can still rely on the interfaces.